File: programmers/js/level00/120883.js
// 로그인 성공?
function solution(id_pw, db) {
  const [inputId, inputPw] = id_pw;
  db = new Map(db);
  
  return db.has(inputId) ? 
    (db.get(inputId) === inputPw ? 'login' : 'wrong pw') :
    'fail';
}

// 다른 풀이
// function solution(id_pw, db) {
//   db = db.filter(v => v[0] === id_pw[0]);

//   if (!db.length) return 'fail';

//   for (let d of db) if (d[1] === id_pw[1]) return 'login';

//   return 'wrong pw';
// }

File: programmers/java/level00/120883.java
// 로그인 성공?

class Solution {
    public String solution(String[] id_pw, String[][] db) {
        String id = id_pw[0];
        String pw = id_pw[1];
        boolean isSameId = false;
        boolean isSamePw = false;
        
        for (String[] idPwInDB : db) {
            String idInDB = idPwInDB[0];
            String pwInDB = idPwInDB[1];
            
            if (id.equals(idInDB)) {
                isSameId = true;
                
                if (pw.equals(pwInDB)) {
                    isSamePw = true;
                }
                
                break;
            }
        }
        
        return isSameId 
            ? isSamePw 
                ? "login"
                : "wrong pw"
            : "fail";
    }
}